aidial_adapter_bedrock/llm/truncate_prompt.py

```diff
@@ -66,15 +66,41 @@ def partition_indexer(chunks: List[int]) -> Callable[[int], List[int]]:
     return mapping.__getitem__
 
 
-T = TypeVar("T")
-Messages = List[T]
+_T = TypeVar("_T")
 
 
-def truncate_prompt(
-    messages: Messages,
-    tokenize_messages: Callable[[Messages], int],
-    keep_message: Callable[[Messages, int], bool],
-    partition_messages: Callable[[Messages], List[int]],
+async def truncate_prompt(
```

Comment:
What is now duplicated, will become triplicated and many-more-times-licated in future, when we add new models with truncate prompt supports. And we will.

Reply:
Ok, can we at least get rid of this `truncate_prompt` and `tuncate_prompt_`?


Imgaine, that you're new to this repo, and you just scrolling codebase, and see two functions, with basically same naming, and different output types

You have two functions
```python
async def truncate_prompt(
...
) -> Tuple[List[int], List[_T]]:



async def truncate_prompt_(
...
) -> Set[int] | TruncatePromptError:
```

I can't imagine any situation, where you're not confused by this, without actually digging into each of this functions

Reply:
Renamed `truncate_prompt_` => `compute_discarded_messages` since it only returns a list of indices of discarded messages, whereas `truncate_prompt` does actually return a truncated list of messages.